python310Packages.django-rapyd-modernauth: init at 0.0.4

greg-hellings opened the pull request:
###### Description of changes

The package is a plugin for Django providing a modern authorization platform.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes (or backporting 23.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

greg-hellings committed:
python310Packages.django-rapyd-modernauth: init at 0.0.4

SuperSandro2000 commented on pkgs/development/python-modules/django-rapyd-modernauth/default.nix:
```diff
@@ -0,0 +1,42 @@
+{
+    lib, buildPythonPackage, fetchPypi,
+
+    django,
+    python-dotenv,
+    pytz,
+    setuptools,
+    sqlparse,
+    zipp
+}:
```
```suggestion
{ lib
, buildPythonPackage
, fetchPypi
, django
, python-dotenv
, pytz
, setuptools
, sqlparse
, zipp
}:
```

SuperSandro2000 commented on pkgs/development/python-modules/django-rapyd-modernauth/default.nix:
```diff
@@ -0,0 +1,42 @@
+}:
+
+buildPythonPackage rec {
+    pname = "django-rapyd-modernauth";
+    version = "0.0.4";
```
```suggestion
  pname = "django-rapyd-modernauth";
  version = "0.0.4";
```
we use 2 spaces for nix files

SuperSandro2000 commented on pkgs/development/python-modules/django-rapyd-modernauth/default.nix:
```diff
@@ -0,0 +1,42 @@
+        sha256 = "sha256-kDZjI32LcKsmLI38ruINKOUfi6lWTIqXF5qIQHi0LeQ=";
+    };
+
+    meta = with lib; {
+        description = "A Django application that provides a custom User model where the username is the email address.";
+        homepage = "https://github.com/karthicraghupathi/django_rapyd_modernauth";
+        license = licenses.afl20;
+        maintainers = [ maintainers.greg ];
+    };
```
meta goes last

SuperSandro2000 commented on pkgs/development/python-modules/django-rapyd-modernauth/default.nix:
```diff
@@ -0,0 +1,42 @@
+        maintainers = [ maintainers.greg ];
+    };
+
+    doCheck = false;
```
Why are tests disabled? We want to run them.

greg-hellings replied:
Because the tests do not get packaged with the PyPI package by upsteam, so they can't be run during our install.

SuperSandro2000 replied:
Can we fetch from github instead?

greg-hellings replied:
Re-attempting with the GH release.

SuperSandro2000 commented:
The commit message prefix should be ``python310Packages.django-rapyd-modernauth``

mweinelt commented on pkgs/top-level/python-packages.nix:
```diff
@@ -2807,6 +2807,8 @@ self: super: with self; {
 
   djangoql = callPackage ../development/python-modules/djangoql { };
 
+  django-rapyd-modernauth = callPackage ../development/python-modules/django-rapyd-modernauth { };
+
   django-ranged-response = callPackage ../development/python-modules/django-ranged-response { };
```
These two need to be swapped.

greg-hellings commented:
OK, I've addressed the comments. The only outstanding one is the tests question. Upstream does not package tests out of the git repo for use from the PyPI package source.

greg-hellings commented:
@SuperSandro2000 Can I get this review updated?

SuperSandro2000 commented on pkgs/development/python-modules/django-rapyd-modernauth/default.nix:
```diff
@@ -0,0 +1,53 @@
+  # Tests are not part of the released package, so we cannot run them
+  doCheck = true;
+  checkPhase = ''
+    set -x
```
```suggestion
```

SuperSandro2000 commented on pkgs/development/python-modules/django-rapyd-modernauth/default.nix:
```diff
@@ -0,0 +1,52 @@
+    hash = "sha256-9IIW6Xsaj2+/SR2CIX4fnUHlAehmLulpBAC7YB3ZrZE=";
+  };
+
+  # Tests are not part of the released package, so we cannot run them
+  doCheck = true;
```
```suggestion
```
thats no longer true

greg-hellings replied:
Indeed, removed.

SuperSandro2000 commented on pkgs/development/python-modules/django-rapyd-modernauth/default.nix:
```diff
@@ -0,0 +1,52 @@
+  # Tests are not part of the released package, so we cannot run them
+  doCheck = true;
+  checkPhase = ''
+    echo 'LOG_LEVEL="INFO"' > .env
+    echo DJANGO_SECRET_KEY="$(base64 /dev/urandom | head -c50)" >> .env
+    python3 ./testrunner.py
```
Could we avoid this by using pytest-django?

greg-hellings replied:
I don't know that we can. The upstream package does not have pytests. It just has some manual commands that get run in the `testrunner.py` main function. It's quite non-standard in its test setup.

SuperSandro2000 replied:
```suggestion
    echo 'LOG_LEVEL="INFO"' > .env
    echo DJANGO_SECRET_KEY="$(base64 /dev/urandom | head -c50)" >> .env
    ${python.interpreter} ./testrunner.py
```
then we want to do this

greg-hellings commented:
OK, I think I have incorporated all of the feedback on the initial version of the PR.

All of the xontribs are moved to `pkgs.xonsh.xontribs.*`.
You can build a custom xonsh with a set of xontribs by using `pkgs.xonsh.withXontribs (x: [ x.xontrib-vox x.xontrib-bashisms])` and similar.
The unwrapped Xonsh package now lives in shells/xonsh/unwrapped.nix

Anyone using the plain `pkgs.xonsh` derivation should not be interrupted, and should still have the functionality of `vox` included. Prior to 0.14.0 vox was included in core Xonsh.

SuperSandro2000 commented on pkgs/development/python-modules/django-rapyd-modernauth/default.nix:
```diff
@@ -0,0 +1,50 @@
+, buildPythonPackage
+, fetchFromGitHub
+, asgiref
+, django
```
```suggestion
, django
, python
```

greg-hellings replied:
Merged in

Artturin commented:
Added missing runHook's and squashed

Artturin commented:
build fails :|

```
python3.10-django-rapyd-modernauth> running install tests
python3.10-django-rapyd-modernauth> base64: write error: Broken pipe
python3.10-django-rapyd-modernauth> base64: write error
python3.10-django-rapyd-modernauth> 2023-09-01 03:48:33,701 CRITICAL testproject Uncaught exception
python3.10-django-rapyd-modernauth> Traceback (most recent call last):
python3.10-django-rapyd-modernauth>   File "/build/source/./testrunner.py", line 33, in <module>
python3.10-django-rapyd-modernauth>     main()
python3.10-django-rapyd-modernauth>   File "/build/source/./testrunner.py", line 18, in main
python3.10-django-rapyd-modernauth>     django.setup()
python3.10-django-rapyd-modernauth>   File "/nix/store/4y4y0y3asww53w03hy5xlplambdam7qp-python3.10-Django-4.2.4/lib/python3.10/site-packages/django/__init__.py", line 24, in setup
python3.10-django-rapyd-modernauth>     apps.populate(settings.INSTALLED_APPS)
python3.10-django-rapyd-modernauth>   File "/nix/store/4y4y0y3asww53w03hy5xlplambdam7qp-python3.10-Django-4.2.4/lib/python3.10/site-packages/django/apps/registry.py", line 116, in populate
python3.10-django-rapyd-modernauth>     app_config.import_models()
python3.10-django-rapyd-modernauth>   File "/nix/store/4y4y0y3asww53w03hy5xlplambdam7qp-python3.10-Django-4.2.4/lib/python3.10/site-packages/django/apps/config.py", line 269, in import_models
python3.10-django-rapyd-modernauth>     self.models_module = import_module(models_module_name)
python3.10-django-rapyd-modernauth>   File "/nix/store/pkj7cgmz66assy7l18zc7j992npb41nx-python3-3.10.12/lib/python3.10/importlib/__init__.py", line 126, in import_module
python3.10-django-rapyd-modernauth>     return _bootstrap._gcd_import(name[level:], package, level)
python3.10-django-rapyd-modernauth>   File "<frozen importlib._bootstrap>", line 1050, in _gcd_import
python3.10-django-rapyd-modernauth>   File "<frozen importlib._bootstrap>", line 1027, in _find_and_load
python3.10-django-rapyd-modernauth>   File "<frozen importlib._bootstrap>", line 1006, in _find_and_load_unlocked
python3.10-django-rapyd-modernauth>   File "<frozen importlib._bootstrap>", line 688, in _load_unlocked
python3.10-django-rapyd-modernauth>   File "<frozen importlib._bootstrap_external>", line 883, in exec_module
python3.10-django-rapyd-modernauth>   File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
python3.10-django-rapyd-modernauth>   File "/build/source/modernauth/models.py", line 3, in <module>
python3.10-django-rapyd-modernauth>     from django.utils.translation import ugettext_lazy as _
python3.10-django-rapyd-modernauth> ImportError: cannot import name 'ugettext_lazy' from 'django.utils.translation' (/nix/store/4y4y0y3asww53w03hy5xlplambdam7qp-python3.10-Django-4.2.4/lib/python3.10/site-packages/django/utils/translation/__init__.py)
```

greg-hellings commented:
I'm going to close this PR rather than leave it sitting. The issue is that the library is targeting older versions of Django. I'm not sure it would be worthwhile for others.
